### DISCUSSION TITLE
Implement lyrics/subtitles

### DISCUSSION THREAD

**Discussion by smileyhead** [ppy/osu#13065] [Ideas]
**Describe the new feature:** This is technically already possible with storyboards, but it would look nicer, and be more versatile and user-friendly this way. It would draw above background dim and support multiple languages.
You should probably also be able to enable and disable lyrics, and set your preferred language in settings and the pre-map quick settings.

**Proposal designs of the feature:** 
![image](https://user-images.githubusercontent.com/15932372/67223096-ee962100-f42e-11e9-9bb3-a624d05b2dd4.png)
![image](https://user-images.githubusercontent.com/15932372/67223338-61070100-f42f-11e9-9e71-f762e7dadfdd.png)

**Comment by bdach** [comment]
Seems awfully large in scope for a relatively small payoff. I'm not sure I would ever care about lyrics during gameplay.

[This custom ruleset could be related/interesting, however.](https://github.com/ppy/osu/issues/5852#issuecomment-525740481)

**Comment by frenzibyte** [comment]
I honestly upvote this as a new storyboard element or similar.

**Comment by smileyhead** [comment]
> Seems awfully large in scope for a relatively small payoff.

There are already people who make lyrics via storyboards and that requires much more time and effort. This would make implementing lyrics much simpler and accessible. I'm willing to bet there would be more maps with lyrics, too.

**Comment by bdach** [comment]
> There are already people who make lyrics via storyboards and that requires much more time and effort. This would make implementing lyrics much simpler and accessible. I'm willing to bet there would be more maps with lyrics, too.

Fair enough, but:

* Just putting the lyrics down as a UI element as in your design takes away artistic license for the storyboard makers (positioning, typography...)
* This would *really* require a non-trivial amount of effort if this was to be implemented entirely within lazer, especially with the scope you've set down in the proposal (ie. in-editor synchronisation, translations).

Internationalisation in general is *never* easy, so caution has to be exercised. I'm not saying this is straight out unworkable, but at the current time I can't imagine it being anything but very low priority.

**Comment by revelsix** [comment]
A simpler form of this idea is just having the ability to add text to storyboards (along with custom fonts / colors) without having to add a png of every letter you want to use, which is especially frustrating for Japanese

**Comment by peppy** [comment]
Not sure about this one. Probably best to leave it to storyboards as a creative option, or the karaoke ruleset as a required feature.

**Comment by andy840119** [comment]
Wait

**Comment by smileyhead** [comment]
> Wait

Hmm?

**Comment by andy840119** [comment]
I have no idea about this because it seems OK
But the text are too small, and not easy to see while playing.
.
(Or can use [my ruleset](https://www.youtube.com/watch?v=3Y6lYVc7LmY) to create karaoke beatmap.)
